lib/commands/sdk.js

```diff
@@ -535,713 +473,575 @@ async function select(logger, config, cli) {
-			str += ex.toString();
-		}
-		throw new Error(str);
-	}
+			const builds = await getBranchBuilds(branch, osName);
+			const build = uri === 'latest' ? builds[0] : builds.find(b => b.name.toLowerCase() === uri);
+			if (!build) {
+				throw new Error(`CI build ${subject} does not exist`);
+			}
 
-	// make sure sdk folder is writable when installing an sdk
-	if (!afs.isDirWritable(installLocation)) {
-		throw new Error(__('Installation location is not writable: %s', installLocation) + '\n');
-	}
-}
+			const asset = build.assets.find(a => a.os === osName);
+			if (!asset) {
+				throw new Error(`CI build ${subject} does not support ${osName}`);
+			}
 
-/**
- * @param {Object} logger - The logger instance
- * @param {Object} config - The CLI config object
- * @param {CLI} cli - The CLI instance
- * @returns {Promise<SDKInstallRequest>}
- */
-async function handleInstallArgs(logger, config, cli) {
-	const version = cli.argv.version;
-	// const branch = cli.argv.branch;
-	const osName = cli.env.os.name;
-
-	if (version) {
-		// version is a URL?
-		if (/^http(s)?:\/\/.+/.test(version)) {
-			return new SDKInstallRequest(logger, config, cli, { url: version });
-		}
+			url = asset.url;
+		} else {
+			// try to find the release by name
+			let release = null;
+
+			if (uri === 'latest') {
+				release = releases.find(r => r.type === 'ga');
+			} else if (uri === 'latest-rc') {
+				release = releases.find(r => r.type === 'rc');
+			} else if (uri === 'latest-beta') {
+				release = releases.find(r => r.type === 'beta');
+			} else {
+				release = releases.find(r => r.name.toLowerCase() === uri);
+				if (!release) {
+					const name = `${uri}.ga`;
+					release = releases.find(r => r.name.toLowerCase() === name);
+				}
+			}
 
-		// version is a filepath?
-		if (/.+\.zip$/.test(version)) {
-			const file = afs.resolvePath(version);
-			if (await fs.pathExists(file)) {
-				const stat = await fs.stat(file);
-				if (stat.isFile()) {
-					const zipFile = await handleGitHubArtifact(file);
-					const manifest = await getZippedManifest(zipFile);
-					return new LocalSDKInstallRequest(logger, config, cli, zipFile, manifest);
+			if (release) {
+				const asset = release.assets.find(a => a.os === osName);
+				if (!asset) {
+					throw new Error(`SDK release ${subject} does not support ${osName}`);
 				}
+				url = asset.url;
 			}
 		}
 
-		// version is a git hash?
-		// const match = version.match(/^([A-Za-z0-9_]+?):(.+)$/);
-		// if (match) {
-		// 	return doBranch(logger, config, cli, match[2], match[1], osName);
-		// }
-	}
+		if (!url) {
+			throw new Error(`Unable to find any Titanium SDK releases or CI builds that match "${subject}"`);
+		}
 
-	// if (branch) {
-	// 	return doBranch(logger, config, cli, version, branch, osName);
-	// }
-	return doReleases(logger, config, cli, version, osName);
-}
+		// step 1.5: download the file
 
-/**
- * Validates the zip that has been requested to be installed. This is because when downloading a
```

Comment:
I think it's good to keep the context of why we look inside the zip for some things